Add SSL certificate

matusik-ops opened the pull request:
MyWeb upgrade about SSL certificate with CertBot and config NGINX.

Commit:
Add SSL certificate

MyWeb upgrade about SSL certificate with CertBot and config NGINX.

Signed-off-by: Martin <matkodevops@gmail.com>

matofeder commented on Ansible/nginx_ssl.conf.j2:
```diff
@@ -0,0 +1,18 @@
+    auth_basic_user_file /etc/nginx/.htpasswd;
+    proxy_pass http://localhost:5000/todo;  # Forward requests to the Flask application
+  }
+}
```
Now you could add an HTTPS (443) nginx config for your web with the path to the certificate. 

Also, ensure that nginx redirects port 80 (HTTP) to 443 (HTTPS)

matofeder commented on Ansible/init_docker/tasks/main.yml:
```diff
@@ -107,13 +109,27 @@
+- name: Restart Nginx
+  ansible.builtin.service:
+    name: nginx
+    state: restarted
```
Isn't `reloaded` sufficient?

matofeder commented on Ansible/init_docker/tasks/main.yml:
```diff
@@ -107,13 +109,27 @@
+    name: nginx
+    state: restarted
+
+
```
two blank lines ;)

Commit:
Fix misstakes and new changes

Create: loop for nginx_ssl.conf.j2, handlers for role, list of domain_name in vars

Signed-off-by: Martin <matkodevops@gmail.com>

matofeder commented on Ansible/init_docker/tasks/main.yml:
```diff
@@ -107,22 +109,28 @@
+- name: Obtain SSL certificate for domain with Certbot
+  ansible.builtin.command: >
+    certbot --nginx -d {{ item }} --email {{ email }} --agree-tos --non-interactive
+  loop: "{{ domain_name}}"
```
It seems the second loop iteration just overrides the certificates from the first one.
I would expect command where both domains are mentions at once ...`-d webma.sk,www.webma.sk`

see output of curl:
```
curl https://webma.sk 
curl: (60) SSL: no alternative certificate subject name matches target host name 'webma.sk'
```

and also check the cert 
```
openssl s_client -connect webma.sk:443 | openssl x509 -noout -text
```

matusik-ops replied:
I fixed the problem. Check it pls. 33687a662cc1937c9726712387ed297c128f9768

matofeder replied:
why not use 

`-d {{ domain_name | join(',') }}`
instead of 
`-d {{ domain }} -d www.{{ domain }}`


Take into account a scenario where you want to add an additional sub-domain, such as todo.webma.sk.

matofeder replied:
also consider using a better name for the variable `domain_name`, e.g. `domains`...

matusik-ops replied:
i fixed the problem...aae25f74b5975ad5750bdaf24ee535c86e5b453f

Commit:
Fix mistakes with task

Add flag to command in task Obtain SSL certificate for domain with Certbot

Signed-off-by: Martin <matkodevops@gmail.com>

Commit:
Change var

Signed-off-by: Martin <matkodevops@gmail.com>
